### PR TITLE
Use OpenAPI client generation for unit tests.

### PR DESCRIPTION
**Pull request by jander-msft**
Generate the APIClient from the OpenAPI document using NSwag; this allows for better verification that the OpenAPI document is usable and functional.

### CODE REVIEW

File: src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests.ApiClient/Microsoft.Diagnostics.Monitoring.UnitTests.ApiClient.csproj [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    </OpenApiReference>
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.ApiDescription.Client" Version="$(MicrosoftExtensionsApiDescriptionClientVersion)">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+    <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersion)" />
+    <PackageReference Include="NSwag.ApiDescription.Client" Version="$(NSwagApiDescriptionClientVersion)">
```
Comment (wiktork) [inline src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests.ApiClient/Microsoft.Diagnostics.Monitoring.UnitTests.ApiClient.csproj:20]: Any known issues with the generated api?

Comment (wiktork) [inline src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests.ApiClient/Microsoft.Diagnostics.Monitoring.UnitTests.ApiClient.csproj:20]: While this is test only code that never gets shipped, do we need to update our 3rd party library files?

**Review by wiktork — APPROVED** [review]

File: src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests.ApiClient/Microsoft.Diagnostics.Monitoring.UnitTests.ApiClient.csproj (hunk 1 above)
Comment (hoyosjs) [reply src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests.ApiClient/Microsoft.Diagnostics.Monitoring.UnitTests.ApiClient.csproj:20]: No, as long as it's just testing assets it doesn't go in the Third Party Notices

**Comment by jander-msft** [comment]
Clsoing. Will continue to hand-write HTTP client for now since the client is not the focus of the tests.

**Pull request closed without merging**